mintscript/src/script/mod.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2024 RBB S.r.l
+// opensource@mintlayer.org
+// SPDX-License-Identifier: MIT
+// Licensed under the MIT License;
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://github.com/mintlayer/mintlayer-core/blob/master/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+pub mod error;
+
+use common::{
+    chain::{
+        signature::{
+            inputsig::standard_signature::StandardInputSignature, sighash::signature_hash,
+            Transactable,
+        },
+        timelock::OutputTimeLock,
+        ChainConfig, Destination, TxOutput,
+    },
+    primitives::H256,
+};
+
+use crate::{
+    helpers::{BlockchainState, SourceBlockState},
+    timelock_check::check_timelock,
+};
+
+#[derive(Debug)]
+pub enum MintScript {
+    Bool(bool),
+    And(Box<MintScript>, Box<MintScript>),
+    Or(Box<MintScript>, Box<MintScript>),
```

Comment:
It think a threshold construct would be more flexible/useful here in place of `And`, `Or`.

Something like:
```suggestion
    Threshold(usize, Vec<MintScript>),
```

* `Threshold(k, conds)` means at least `k` of the specified conditions have to be satisfied
* N-ary conjunction (and) becomes N-of-N threshold
* N-ary disjunction (or) becomes 1-of-N threshold
* `And`, `or` may be given a constructor function for convenience.
* Also allows for convenient encoding of the likes of 3-of-5 multisig and such which are rather awkward to expsess using `and`s / `or`s.

Reply:
Done. Thanks for the suggestion.